src/Protocol/Diaspora.php

```diff
@@ -511,77 +518,77 @@ public static function dispatch(array $importer, array $msg, SimpleXMLElement $f
+		Logger::info('Received message', ['type' => $type, 'sender' => $sender, 'user' => $importer['uid']]);
 
 		switch ($type) {
-			case "account_migration":
+			case 'account_migration':
 				if (!$private) {
 					Logger::notice('Message with type ' . $type . ' is not private, quitting.');
 					return false;
 				}
 				return self::receiveAccountMigration($importer, $fields);
 
-			case "account_deletion":
+			case 'account_deletion':
 				return self::receiveAccountDeletion($fields);
 
-			case "comment":
-				return self::receiveComment($importer, $sender, $fields, $msg["message"], $direction);
+			case 'comment':
+				return self::receiveComment($importer, $sender, $fields, $msg['message'], $direction);
 
-			case "contact":
+			case 'contact':
 				if (!$private) {
 					Logger::notice('Message with type ' . $type . ' is not private, quitting.');
 					return false;
 				}
 				return self::receiveContactRequest($importer, $fields);
 
-			case "conversation":
+			case 'conversation':
 				if (!$private) {
 					Logger::notice('Message with type ' . $type . ' is not private, quitting.');
 					return false;
 				}
 				return self::receiveConversation($importer, $msg, $fields);
 
-			case "like":
+			case 'like':
 				return self::receiveLike($importer, $sender, $fields, $direction);
 
-			case "message":
+			case 'message':
 				if (!$private) {
 					Logger::notice('Message with type ' . $type . ' is not private, quitting.');
 					return false;
 				}
 				return self::receiveMessage($importer, $fields);
 
-			case "participation":
+			case 'participation':
 				if (!$private) {
 					Logger::notice('Message with type ' . $type . ' is not private, quitting.');
 					return false;
 				}
 				return self::receiveParticipation($importer, $fields, $direction);
 
-			case "photo": // Not implemented
+			case 'photo': // Not implemented
 				return self::receivePhoto($importer, $fields);
 
-			case "poll_participation": // Not implemented
+			case 'poll_participation': // Not implemented
 				return self::receivePollParticipation($importer, $fields);
 
-			case "profile":
+			case 'profile':
 				if (!$private) {
 					Logger::notice('Message with type ' . $type . ' is not private, quitting.');
 					return false;
 				}
 				return self::receiveProfile($importer, $fields);
 
-			case "reshare":
-				return self::receiveReshare($importer, $fields, $msg["message"], $direction);
+			case 'reshare':
+				return self::receiveReshare($importer, $fields, $msg['message'], $direction);
 
-			case "retraction":
+			case 'retraction':
 				return self::receiveRetraction($importer, $sender, $fields);
 
-			case "status_message":
-				return self::receiveStatusMessage($importer, $fields, $msg["message"], $direction);
+			case 'status_message':
+				return self::receiveStatusMessage($importer, $fields, $msg['message'], $direction);
 
 			default:
-				Logger::notice("Unknown message type ".$type);
+				Logger::notice("Unknown message type " . $type);
```

Comment:
Since we're going the full conversion route, let's pick up stragglers.
```suggestion
				Logger::notice('Unknown message type ' . $type);
```

Reply:
Even better:
```
DI::logger->notice('Unknown message type', 'type' => $type);
```

So we're marking the Dependency Injection directly with 'DI' and we use parameter so we can distinct between message and payload of an log entry. This is helpful when we use enterprise logging systems like Splunk/Elastic

Reply:
@nupplaphil A lot lines then needs rework. I would find it more proper to move this to another `rewrite/log-messages` branch/PR instead of doing it here.

The other question that came up here is, when it is a unknown (e.g. not yet handled) message-type, I would find a warning proper than just a notice as maybe some `case 'foo':` is missing here then. But that's only how I would do it.

Reply:
@nupplaphil I agree with you, but it isn't in the scope of this PR that's already humongous.